### ISSUE TITLE
Expanding folder saves all files inside unnecessarily

### ISSUE THREAD

**Issue by benhamlin** [Microsoft/vscode#36653]
- VSCode Version: Code - Insiders 1.18.0-insider (82dcf9835265cd0a45ec135587ae2a82673f1c8f, 2017-10-20T04:24:25.632Z)
- OS Version: Windows_NT x64 10.0.17020
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
Handlebars|and|0.2.0
vsc-ember-frost|cie|0.0.2
jshint|dba|0.10.15
gitlens|eam|5.7.1
PowerShell|ms-|1.4.3
debugger-for-chrome|msj|3.4.0
vscode-json-transform|oct|0.1.2
vscode-icons|rob|7.15.0
code-spell-checker|str|1.4.10
html-css-class-completion|Zig|1.11.0

---

Steps to Reproduce:

1. Use some kind of file watcher, live reload, etc type tool
2. Expand a folder in the VSCode file view
3. Note that all/some of the files are saved

This isn't a huge problem since my live build is < 5sec but if it was longer it could be pretty annoying. It only seems to happen the first time the folder is expanded, like VSCode is caching/analysing/proccessing the files somehow? There are no actual changes to the files, they just get touched. 

Disabled all extensions to check, still repros.

Reproduces without extensions: Yes

**Comment by isidorn** [comment]
@bpasero do we do something special when we resolve files for the first time?

**Comment by bpasero** [comment]
@benhamlin what file watcher are you using? It could well be that this file watcher fires an event when a file is accessed which could happen when we read the metadata of the files.

@isidorn no, just reading

**Comment by benhamlin** [comment]
@bpasero ember-cli 2.14, specifically 'ember build --watch'.

The LastAccessTime on the files is not changing, if that's what you mean. 'fsutil behavior query DisableLastAccess' returns 1 for me, so reading files doesn't write them. 

I've tested other editors (notepad, Visual Studio) and that doesn't trigger ember when I open files (let alone list them), so it does appear to be something unique to VSCode.

On a hunch, I copied my project into another folder and deleted the .git folder, and there the problem no longer repros. So it might have something to do with how VSCode integrates with git.

**Comment by isidorn** [comment]
fyi @joaomoreno

**Comment by joaomoreno** [comment]
@benhamlin Does running `git status` touch cause the same effect?

**Comment by benhamlin** [comment]
@joaomoreno nope, afraid not.

**Comment by joaomoreno** [comment]
So it isn't VSCode's git integration, since that simply calls `git status` in the background.

**Comment by isidorn** [comment]
@benhamlin sorry for the slow response but can you still reproduce this with latest vscode version?

**Comment by benhamlin** [comment]
Yes, I'm still seeing the behavior in 1.23.0.

In the time since I opened this, I've completely switched toolsets, so now the file watchers I'm using that are being triggered are webpack-dev-server and jest. Interestingly, the two tools are not always both triggered. Jest seems to be the most sensitive, and triggers fresh test runs almost every time I open a folder or file for the first time. webpack-dev-server seems far more discerning, although I was able to trigger a rebuild a couple times by opening folders.

This does seem to imply that at least some of the blame falls on the file watching tools themselves. So this may not be VSCode's fault. But I still can't repro this outside of VSCode...
